crates/router/src/connector/cryptopay/transformers.rs

```diff
@@ -155,14 +155,24 @@ impl<F, T>
             types::PaymentsResponseData,
         >,
     ) -> Result<Self, Self::Error> {
-        let redirection_data = item
-            .response
-            .data
-            .hosted_page_url
-            .map(|x| services::RedirectForm::from((x, services::Method::Get)));
-        Ok(Self {
-            status: enums::AttemptStatus::from(item.response.data.status),
-            response: Ok(types::PaymentsResponseData::TransactionResponse {
+        let status = enums::AttemptStatus::from(item.response.data.status.clone());
+        let response = if is_payment_failure(status) {
+            let payment_response = &item.response.data;
+            Err(types::ErrorResponse {
+                code: payment_response.name.clone().unwrap_or_default(),
```

Comment:
Don't use `unwrap_or_default()` instead use `unwrap_or(consts::NO_ERROR_CODE.to_string())` for error code and `unwrap_or(consts::NO_ERROR_MESSAGE.to_string())` for message